### ISSUE TITLE
Assertion on getCookie() is called twice in the Cypress runner

### ISSUE THREAD

**Issue by abhidp** [cypress-io/cypress#6672]
### Current behavior:
I am asserting `cy.getCookie()` only once in my code, but the Cypress runner shows the assertion happened twice. Though it is a very minor and low priority issue, it is not the most optimum as the logic for validating the cookie from `cypress-runner.js` is being called twice and adds to the execution time. 

### Desired behavior:

Cypress runner should show the `ASSERT` log only once for `cy.getCookie('auth').should('exist')`

### Test code to reproduce

```javascript
// cypress/integration/login.spec.js
it('should login successfully with correct credentials and get auth cookie', () => {
    cy.get('#loginForm').within(() => {
      cy.get('#email')
        .clear()
        .type('cypress@crate.com')
        .get('#password')
        .clear()
        .type('Pass1234');
      cy.contains('Login').click();
    });
    cy.title().should('eq', 'Crates for everyone! - Crate');
    cy.url().should('eq', `${Cypress.config().baseUrl}/crates`);
    cy.getCookie('auth').should('exist');
  });
```

![image](https://user-images.githubusercontent.com/30851622/76138018-7eafcc80-6097-11ea-82ad-a48658bc8b44.png)

Both the `ASSERT`s print exactly the same message from the same line number of the Cypress runner source code `cypress-runner.js:168422` to the DevTools console

![image](https://user-images.githubusercontent.com/30851622/76138078-19a8a680-6098-11ea-93ee-5b70c4f8d07d.png)

![image](https://user-images.githubusercontent.com/30851622/76138067-08f83080-6098-11ea-9de9-15599317e6fd.png)

### Versions

Cypress: 

```json
"devDependencies": {
    "cypress": "^4.1.0"
 }
```

OS:

```
ProductName:	Mac OS X
ProductVersion:	10.14.6
```

Browser:

```
Google Chrome is up to date
Version 80.0.3987.132 (Official Build) (64-bit)
```

**Comment by jennifer-shehane** [comment]
Yeah...not sure why it's doing that. 

This began happening in [Cypress 4.0.0](https://on.cypress.io/changelog#4-0-0). There's a lot of changes in 4.0.0, but my complete guess is maybe it involves the Chai upgrade https://github.com/cypress-io/cypress/pull/2862.

`spec.js`

```js
it('cookie verification', () => {
  cy.visit('https://example.cypress.io/commands/cookies')
  cy.get('#getCookie .set-a-cookie').click()
  cy.getCookie('token').should('exist')
});
```

<img width="521" alt="Screen Shot 2020-03-09 at 9 27 14 PM" src="https://user-images.githubusercontent.com/1271364/76226825-cdeb2e00-624c-11ea-94db-7f4eaeb81e9f.png">

<img width="519" alt="Screen Shot 2020-03-09 at 9 41 14 PM" src="https://user-images.githubusercontent.com/1271364/76228144-ba40c700-624e-11ea-8328-ed99a8a91206.png">


The below code does *not* log the assertion twice. 

```js
it('cookie verification', () => {
  cy.visit('https://example.cypress.io/commands/cookies')
  cy.get('#getCookie .set-a-cookie').click()
  cy.getCookie('token').then((cookie) => {
    expect(cookie).to.exist
  })
});
```

**Comment by Tomastomaslol** [comment]
HI @jennifer-shehane, 

I had a look at this and it looks like the change in https://github.com/cypress-io/cypress/pull/2862 that is causing the assertion to run twice is:

https://github.com/cypress-io/cypress/pull/2862/files#diff-462c090a4a1cf87c581bae15f3af0d6aR111

It looks the same behaviour has been ported from the Coffescript to the Javascript version of `asserting`.

**Example spec**
```
it('cookie verification', () => {
  cy.visit('https://example.cypress.io/commands/cookies')
  cy.get('#getCookie .set-a-cookie').click()
  cy.getCookie('token').should('exist')
});
```
### Running  the spec against latest develop

![Screen Shot 2020-08-11 at 1 02 48 pm](https://user-images.githubusercontent.com/1872246/89855201-3d3cc380-dbd9-11ea-9c39-49018a03f10c.png)


### Running the spec against https://github.com/cypress-io/cypress/pull/8246

![Screen Shot 2020-08-11 at 12 57 51 pm](https://user-images.githubusercontent.com/1872246/89855274-6cebcb80-dbd9-11ea-9f88-c8f9e95f982d.png)


I don't really understand why the code was introduced in the first place but it looks like most checks pass against the draft pull request I opened. Is there anything else you would like me to do or look at before requesting a review for the pull request?

**Comment by jennifer-shehane** [comment]
@Tomastomaslol The PR will require tests to ensure that this is no longer logging twice in the UI. There's some idea of what these tests look like around command logs here: https://github.com/cypress-io/cypress/blob/develop/packages/driver/cypress/integration/commands/cookies_spec.js#L174:L174

**Comment by Tomastomaslol** [comment]
@jennifer-shehane Thanks for your pointer. I opened a pull request that should fix the described issue. If you have a spare minute please let me know what you think of my proposed solution.

 I really enjoyed working with the codebase and hopefully there will be other opportunities for me to contribute in the future!

Have a good Friday! 🙂
